src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -2781,21 +2954,40 @@ HRESULT CProfilerManager::InitializeForAttach(
 
     m_bAttach = true;
 
+    if (pvClientData == nullptr ||
+        cbClientData == 0)
+    {
+        // No configuration provided
+        return S_FALSE;
+    }
+
     vector<CComPtr<CConfigurationSource>> configSources;
 
-    // TODO: Parse configuration information from pvClientData.
+    UINT bufferSize = cbClientData + 1; // Hold data + null terminator
+    //WCHAR* wszConfigXml = new WCHAR[bufferSize]; 
+    unique_ptr<WCHAR[]> wszConfigXml = make_unique<WCHAR[]>(bufferSize);
+
+    LPCWSTR pData = reinterpret_cast<LPCWSTR>(pvClientData);
+    IfFailRetErrno(memcpy_s(wszConfigXml.get(), bufferSize, pData, cbClientData));
+
+    m_wszConfigXml = wszConfigXml.release();
+    m_cbConfigXml = cbClientData;
+
+    IfFailRet(InvokeThreadRoutine(configSources, InitializeForAttachThreadProc));
 
     // Mark that this is during the initialize call. This enables operations that can only be supported during initialize
     CInitializeHolder initHolder(this);
 
-    IfFailRet(InitializeCore(pCorProfilerInfoUnk, configSources));
+    // m_configSources should be populated by InitializeForAttachThreadProc.
+    IfFailRet(InitializeCore(pCorProfilerInfoUnk, m_configSources));
 
     if (m_bProfilingDisabled)
     {
         return CORPROF_E_PROFILER_CANCEL_ACTIVATION;
     }
 
-    IfFailRet(SendEventToRawProfilerCallback(&ICorProfilerCallback3::InitializeForAttach, pCorProfilerInfoUnk, pvClientData, cbClientData));
+    // TODO: willxie The ClientData for RawProfilerHook should be a subset of the ClientData parameter (which includes instrumentationmethods).
+    IfFailRet(SendEventToRawProfilerCallback(&ICorProfilerCallback3::InitializeForAttach, pCorProfilerInfoUnk, (void*)nullptr, 0U));
```

Comment:
Yep, otherwise it thinks nullptr is of type `std::nullptr_t`

---
In reply to: [385970264](https://github.com/microsoft/CLRInstrumentationEngine/pull/216#discussion_r385970264) [](ancestors = 385970264)